### PR TITLE
fix(HACBS-966): Loop over found bindings to filter based on application

### PR DESCRIPTION
**Pull request by dirgim**
Currently the [findExistingApplicationSnapshotEnvironmentBinding](https://github.com/redhat-appstudio/integration-service/blob/main/controllers/integration/integration_adapter.go#L418) function uses mutliple client.MatchingFields calls in order to find bindings that reference the specified environment as well as the application.

Unfortunately, due to the limitation of the FieldSelector filtering when used with cache-based implementations, this approach does not work correctly, and only the `spec.environment` is taken into account. This change does the following:

* findExistingApplicationSnapshotEnvironmentBinding refactored to use a single MatchingFields
* Use a for loop to filter based on application
* Fix related warnings raised by the IDE

Signed-off-by: dirgim <kpavic@redhat.com>

### CODE REVIEW

**Comment by johnbieren** [comment]
lgtm!

**Review by Dannyb48 — APPROVED** [review]
LGTM

File: controllers/integration/integration_adapter.go [hunk 1]
```diff
@@ -489,7 +494,7 @@ func (a *Adapter) EnsureApplicationSnapshotEnvironmentBindingExist() (results.Op
 }
 
 func (a *Adapter) getApplicationSnapshotComponents(components []hasv1alpha1.Component) *[]appstudioshared.BindingComponent {
-	bindingComponents := []appstudioshared.BindingComponent{}
+	var bindingComponents []appstudioshared.BindingComponent
```
Comment (jinqi7) [inline controllers/integration/integration_adapter.go:497]: Can you please make clear - any problem with the origin way?  Or, it's recommended to use the changed way in some condition? Thanks

Comment (dirgim) [reply controllers/integration/integration_adapter.go:497]: I applied the suggested approach after I got a warning from the IDE about `Empty slice declaration using a literal`. After some digging, I've found that this warning can be ignored ([jetbrains ticket here](https://youtrack.jetbrains.com/issue/GO-5317)), so I've decided to revert these changes to use your original approach.

Comment (jinqi7) [reply controllers/integration/integration_adapter.go:497]: Thanks for your answer!

Comment (dirgim) [reply controllers/integration/integration_adapter.go]: @jinqi7 I reverted the mentioned changes, please take a look

Comment (jinqi7) [reply controllers/integration/integration_adapter.go]: OK.

**Comment by jinqi7** [comment]
LGTM

**Review by jinqi7 — APPROVED** [review]

**Review by hongweiliu17 — APPROVED** [review]
LGTM

**Pull request merged**